Source/SwiftLintFramework/Rules/MultilineFunctionChainsRule.swift

```diff
@@ -0,0 +1,221 @@
+    }
+
+    private static let whitespaceDotRegex = regex("\\s*\\.")
+
+    private func callDotOffset(file: File, callRange: NSRange) -> Int? {
+        guard
+            let range = file.contents.bridge().byteRangeToNSRange(start: callRange.location, length: callRange.length),
+            case let regex = type(of: self).whitespaceDotRegex,
+            let match = regex.matches(in: file.contents, options: [], range: range).last?.range else {
+                return nil
+        }
+        return match.location + match.length - 1
+    }
+
+    private static let newlineWhitespaceDotRegex = regex("\\n\\s*\\.")
+
+    private func callHasLeadingNewline(file: File, callRange: NSRange) -> Bool {
+        guard
+            let range = file.contents.bridge().byteRangeToNSRange(start: callRange.location, length: callRange.length),
+            case let regex = type(of: self).newlineWhitespaceDotRegex,
+            regex.firstMatch(in: file.contents, options: [], range: range) != nil else {
+                return false
+        }
+        return true
+    }
+
+    private func callRanges(file: File,
+                            kind: SwiftExpressionKind,
+                            dictionary: [String: SourceKitRepresentable],
+                            parentCallName: String? = nil) -> [NSRange] {
+        guard
+            kind == .call,
+            case let contents = file.contents.bridge(),
+            let offset = dictionary.nameOffset,
+            let length = dictionary.nameLength,
+            let name = contents.substringWithByteRange(start: offset, length: length) else {
+                return []
+        }
+
+        let subcalls = dictionary.subcalls
+
+        if subcalls.isEmpty, let parentCallName = parentCallName, parentCallName.starts(with: name) {
+            return [NSRange(location: offset, length: length)]
+        }
+
+        return subcalls.flatMap { call -> [NSRange] in
+            guard let range = callRange(file: file, call: call, parentName: name, parentNameOffset: offset) else {
+                return []
+            }
+
+            return [range] + callRanges(file: file, kind: .call, dictionary: call, parentCallName: name)
+        }
+    }
+
+    private func callRange(file: File,
+                           call: [String: SourceKitRepresentable],
+                           parentName: String,
+                           parentNameOffset: Int) -> NSRange? {
+        guard
+            case let contents = file.contents.bridge(),
+            let nameOffset = call.nameOffset,
+            parentNameOffset == nameOffset,
+            let nameLength = call.nameLength,
+            let bodyOffset = call.bodyOffset,
+            let bodyLength = call.bodyLength,
+            let name = contents.substringWithByteRange(start: nameOffset, length: nameLength),
+            parentName.starts(with: name) else {
+                return nil
+        }
+
+        let linkOffset = nameOffset + nameLength
```

Comment:
what is a link? 🤔

Reply:
Was thinking like the "link" of a chain here, but I'll make these more descriptive 👍